src/OpenIddict.Core/OpenIddictController.cs

```diff
@@ -228,4 +266,50 @@ public class OpenIddictController<TUser, TApplication> : Controller where TUser
             await HttpContext.Authentication.SignOutAsync(Options.AuthenticationScheme);
         }
     }
+
+    public class ScopeModel
+    {
+        public string ScopeId { get; set; }
+        public string DisplayName { get; set; }
+        public string Description { get; set; }
+    }
+
+    public static class OpenIdConnectMessageExtension
+    {
+        /// <summary>
+        /// Extracts the standard OpenId connect scopes from an <see cref="OpenIdConnectMessage"/>.
+        /// </summary>
+        /// <param name="message">The <see cref="OpenIdConnectMessage"/> instance.</param>
+        public static IEnumerable<string> GetStandardScopes(this OpenIdConnectMessage message)
+        {
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            return message.GetScopes().Intersect(GetOpenIdStandardScopes());
+        }
+
+        /// <summary>
+        /// Extracts the non-standard scopes from an <see cref="OpenIdConnectMessage"/>.
+        /// </summary>
+        /// <param name="message">The <see cref="OpenIdConnectMessage"/> instance.</param>
+        public static IEnumerable<string> GetCustomScopes(this OpenIdConnectMessage message)
+        {
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            return message.GetScopes().Except(GetOpenIdStandardScopes());
+        }
+
+        public static IEnumerable<string> GetOpenIdStandardScopes()
+        {
+            yield return OpenIdConnectConstants.Scopes.OpenId;
+            yield return OpenIdConnectConstants.Scopes.Email;
```

Comment:
Do we need to provide a standard description/display name to display the standard openid connect scopes ```email```, ```profile```, ```offline_access``` (and ```phone```/```address```) in the shiny consent screen? :)